src/mode.js

```diff
@@ -60,15 +60,18 @@ export function getMode(opt_win) {
  * @return {!ModeDef}
  */
 function getMode_(win) {
+  const AMP_CONFIG = self.AMP_CONFIG || {};
+
   // Magic constants that are replaced by closure compiler.
   // IS_MINIFIED is always replaced with true when closure compiler is used
   // while IS_DEV is only replaced when `gulp dist` is called without the
   // --fortesting flag.
   const IS_DEV = true;
   const IS_MINIFIED = false;
 
-  const localDevEnabled = !!(self.AMP_CONFIG && self.AMP_CONFIG.localDev);
-  const runningTests = IS_DEV && !!(win.AMP_TEST || win.__karma__);
+  const localDevEnabled = !!AMP_CONFIG.localDev;
+  const runningTests = (!!AMP_CONFIG.test) || (
+    IS_DEV && !!(win.AMP_TEST || win.__karma__));
```

Comment:
We should further simplify this logic at some point (not in this PR). @erwinmombay should have a good idea of the edge cases we were trying to solve for in the past, and which of them are relevant today. Maybe add a TODO here?

Reply:
Done